src/fraction.rs

```diff
@@ -0,0 +1,184 @@
+use core::{
+    any::type_name,
+    fmt,
+    marker::PhantomData,
+    ops::{Div, Mul},
+};
+
+use typenum::{UInt, Unsigned, U0, U1};
+
+use crate::{eq::FractionEq, from_int::FromUnsigned};
+
+/// Fraction `Numerator / Denominator`
+pub struct Fraction<Numerator, Denominator>(PhantomData<(Numerator, Denominator)>);
+
+/// Default fraction. `1/1`
+pub type One = Fraction<U1, U1>;
+
+impl<N, D> Fraction<N, D> {
+    /// Create new fraction
+    #[inline]
+    pub const fn new() -> Self {
+        Self(PhantomData)
+    }
+}
+
+impl<N, D> Default for Fraction<N, D> {
+    #[inline]
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+///
+pub trait FractionTrait {
+    ///
+    type Numerator: Unsigned;
+
+    ///
+    type Divisor: Unsigned;
+
+    /// Multiply integer by this fraction
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// use typed_phy::{fraction::FractionTrait, Frac};
+    /// use typenum::{U5, U7};
+    ///
+    /// assert_eq!(<Frac![U5 / U7]>::mul(14), 10)
+    /// ```
+    #[inline]
+    fn mul<I>(int: I) -> I
+    where
+        I: FromUnsigned + Mul<Output = I> + Div<Output = I>,
```

Comment:
Probably you want to implement multiplication/division resulting in `f64`/`f32`?

Reply:
Sure. I forgot about `FromUnsigned` impls for `f64`/`f32`.